scripts/ecg_learning.py

```diff
@@ -0,0 +1,103 @@
+import pdb
+import json
+import requests
+import os
+import argparse
+import findspark
+
+from datetime import datetime, timedelta
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, StringType, IntegerType
+from pathlib import Path
+from pyspark.sql import functions as F
+
+from utils import create_json, write_data_to_blob, get_data_from_blob
+
+priometheus_host = os.environ['PROMETHEUS_HOST']
+findspark.init()
+
+def parsing_necessary_details(from_time, to_time):
+    url = "{}/prometheus/api/v1/query_range".format(priometheus_host)
+    querystring = {"query": "sum(rate(nginx_request_status_count{cluster=~\"Swarm1|Swarm2\"}[5m]))","start":str(from_time),"end":str(to_time),"step":"900"}
+    headers = {
+        'Accept': "application/json, text/plain, */*",
+    }
+
+    response = requests.request("GET", url, headers=headers, params=querystring)
+
+    return response.json()["data"]["result"][0]["values"]
+
+
+def remove_last_day(df):
+    current_hour = current_time.hour
+    if current_hour == 0 & df.count() >= 672:
+            first_date = (current_time + timedelta(days=-8)).strftime("%Y/%m/%d")
+            df = df.filter(~F.col("time").contains(first_date))
+
+    return df
+
+def init(from_time, to_time):
+    ecg_data = parsing_necessary_details(from_time, to_time)
+
+    spark = SparkSession.builder.appName("ECGLearning").master("local[*]").getOrCreate()
+
+    os.makedirs(os.path.join(write_path, 'public'), exist_ok=True)
+
+    # Create data frame
+    ecg_data_rdd = spark.sparkContext.parallelize(ecg_data)
+    schema = StructType([
+        StructField('time', IntegerType(), True),
+        StructField('tps', StringType(), True)
+    ])
+
+    tps_df = spark.createDataFrame(ecg_data_rdd,schema)
+
+    tps_df = tps_df.withColumn("tps", tps_df["tps"].cast("float"))
+    tps_df = tps_df.withColumn("tps", F.ceil(tps_df["tps"]))
+    tps_df = tps_df.withColumn("time", F.from_unixtime(tps_df["time"], "yyyy/MM/dd hh:mm:ss"))
```

Comment:
Do we need to add timezone information here? if it is running in a server configured on UTC, will the time be shown in UTC?

Reply:
I used the server's timezone. In the new commit, I have updated the script for the Indian timezone also.